cpp/src/io/parquet/page_decode.cuh

```diff
@@ -954,21 +956,11 @@ struct all_types_filter {
 };
 
 /**
- * @brief Functor for setupLocalPageInfo that returns true if this is not a string column.
- */
-struct non_string_filter {
-  device_span<ColumnChunkDesc const> chunks;
-
-  __device__ inline bool operator()(PageInfo const& page) { return !is_string_col(page, chunks); }
-};
-
-/**
- * @brief Functor for setupLocalPageInfo that returns true if this is a string column.
+ * @brief Functor for setupLocalPageInfo that takes a mask of allowed types.
  */
-struct string_filter {
-  device_span<ColumnChunkDesc const> chunks;
-
-  __device__ inline bool operator()(PageInfo const& page) { return is_string_col(page, chunks); }
+struct mask_filter {
+  int mask;
+  __device__ inline bool operator()(PageInfo const& page) { return (page.kernel_mask & mask) != 0; }
```

Comment:
`setupLocalPageInfo` docs below are a bit broken - return value description is missing, as well as description of the last parameter. Not a request for this PR, more of a note.

Reply:
There was a lot of docstring cruft in this file, so I cleaned up most of it.